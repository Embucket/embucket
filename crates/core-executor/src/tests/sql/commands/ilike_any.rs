use crate::test_query;

test_query!(
    ilike_any_basic,
    "SELECT * FROM VALUES
        ('jane doe'),
        ('Jane Doe'),
        ('JANE DOE'),
        ('John Doe'),
        ('John Smith')
     WHERE column1 ILIKE ANY ('jane%', '%SMITH')
     ORDER BY column1",
    snapshot_path = "ilike_any"
);

test_query!(
    ilike_any_even,
    "SELECT * FROM VALUES
                ('John  Dddoe'),
                ('Joe   Doe'),
                ('John_down'),
                ('Joe down'),
                ('Tom   Doe'),
                ('Tim down'),
                (NULL)
            WHERE column1 ILIKE ANY ('%Jo%oe%','T%e')
            ORDER BY column1",
    snapshot_path = "ilike_any"
);

test_query!(
    ilike_any_odd,
    "SELECT * FROM VALUES
                ('John  Dddoe'),
                ('Joe   Doe'),
                ('John_down'),
                ('Joe down'),
                ('Tom   Doe'),
                ('Tim down'),
                (NULL)
            WHERE column1 ILIKE ANY ('%Jo%oe%','T%e', '%Tim%')
            ORDER BY column1",
    snapshot_path = "ilike_any"
);

test_query!(
    ilike_any_odd_wrong,
    "SELECT * FROM VALUES
                ('John  Dddoe'),
                ('Joe   Doe'),
                ('John_down'),
                ('Joe down'),
                ('Tom   Doe'),
                ('Tim down'),
                (NULL)
            WHERE column1 ILIKE ANY ('%Jo%oe%','T%e', '%Tim%', '%YES%')
            ORDER BY column1",
    snapshot_path = "ilike_any"
);

test_query!(
    ilike_any_even_wrong,
    "SELECT * FROM VALUES
                ('John  Dddoe'),
                ('Joe   Doe'),
                ('John_down'),
                ('Joe down'),
                ('Tom   Doe'),
                ('Tim down'),
                (NULL)
            WHERE column1 ILIKE ANY ('%Jo%oe%', '%YES%')
            ORDER BY column1",
    snapshot_path = "ilike_any"
);

test_query!(
    ilike_any_one,
    "SELECT * FROM VALUES
                ('John  Dddoe'),
                ('Joe   Doe'),
                ('John_down'),
                ('Joe down'),
                ('Tom   Doe'),
                ('Tim down'),
                (NULL)
            WHERE column1 ILIKE ANY ('%Jo%oe%')
            ORDER BY column1",
    snapshot_path = "ilike_any"
);

test_query!(
    ilike_any_one_wrong,
    "SELECT * FROM VALUES
                ('John  Dddoe'),
                ('Joe   Doe'),
                ('John_down'),
                ('Joe down'),
                ('Tom   Doe'),
                ('Tim down'),
                (NULL)
            WHERE column1 ILIKE ANY ('%YES%')
            ORDER BY column1",
    snapshot_path = "ilike_any"
);
