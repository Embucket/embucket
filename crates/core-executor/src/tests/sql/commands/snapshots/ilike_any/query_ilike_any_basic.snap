---
source: crates/core-executor/src/tests/sql/commands/ilike_any.rs
description: "\"SELECT * FROM VALUES\n        ('jane doe'),\n        ('Jane Doe'),\n        ('JANE DOE'),\n        ('John Doe'),\n        ('John Smith')\n     WHERE column1 ILIKE ANY ('jane%', '%SMITH')\n     ORDER BY column1\""
---
Ok(
    [
        "+------------+",
        "| column1    |",
        "+------------+",
        "| JANE DOE   |",
        "| Jane Doe   |",
        "| John Smith |",
        "| jane doe   |",
        "+------------+",
    ],
)
